### DISCUSSION TITLE
Event to create Enum?

### DISCUSSION THREAD

**Discussion by spolloni** [sqlalchemy/sqlalchemy#10186] [Usage Questions] [answered]
Hi folks! I would like to write an event to create an Enum tyoe  in my postgres DB. The reason is that this enum is not explicitly used as a column type so it will not be created when `Metadata.create_all()` is called.  Is it possible to obtain the DDL somehow from the Enum instance?

```
import enum
import sqlalchemy
from sqlalchemy import Enum
from .base import Base

class Schrodinger(str, enum.Enum)
    DEAD = 'DEAD'
    ALIVE = 'ALIVE'

schrodinger_enum = Enum(Schrodinger)

sqlalchemy.event.listen(
    Base.metadata,
    'after_create',
    ???? 
)
```

hoping that I can call something like `schrodinger_enum.to_sql_create(postgresql.dialect())`

thanks!

**Comment by CaselIT** [comment, marked as answer]
Hi,

I don't think there is an actual ddl object that can be executed, but an enum has a create method that you can call in the event https://docs.sqlalchemy.org/en/20/core/type_basics.html#sqlalchemy.types.Enum.create

**Reply by spolloni** [reply]
Thank you @CaselIT, This works.  

Just FYI my final solution ended up simply making sure the type is registered with the metadata, like so:

```
schrodinger_enum = Enum(Schrodinger, metadata=Base.metadata)
```

this entirely addresses the problem I laid out here:

> this enum is not explicitly used as a column type so it will not be created when `Metadata.create_all()` is called

and an event is not even required.

**Reply by CaselIT** [reply]
forgot about Enum.metadata, good find
